### PR TITLE
Resolve `Hip17UnhappyAccountsSuite` errors

### PR DESCRIPTION
**Pull request by natanasow**
**Description**:

We have the following test suite `Hip17UnhappyAccountsSuite` that has problems when it's running against the modularization codebase.

**Solution**:

Resolve issues with the test suite. When ready add @HapiTest annotations to the passing tests.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by natanasow** [comment]
Currently, the system tasks (e.g. for account expiration) are not running when the tests are executed, so these two tests can not be completed:
- `uniqueTokenOperationsFailForExpiredAccount`
- `uniqueTokenOperationsFailForAutoRemovedAccount`

**Comment by natanasow** [comment]
@netopyr Would you take a look :eyes:? That's a tiny one but most of the unit tests are needed for another PR.

**Comment by netopyr** [comment]
I looked at it already, but I think somebody more familiar with `TokenService` needs to review it.

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenGrantKycToAccountHandlerTest.java [hunk 1]
```diff
@@ -221,16 +234,20 @@ void nullAccountIdThrowsException() {
         }
 
         @Test
-        @DisplayName("When getForModify returns empty, should not put or commit")
+        @DisplayName("When TokenRelStore.get() returns empty, should not put or commit")
         void emptyGetForModifyShouldNotPersist() {
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenGrantKycToAccountHandlerTest.java:238]: Function name should match display name modification here as well
```suggestion
        @DisplayName("When TokenRelStore.get() returns empty, should not put or commit")
        void emptyGetShouldNotPersist() {
```

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java [hunk 2]
```diff
@@ -236,10 +245,6 @@ void tokenRelNotFound() throws HandleException {
         @Test
         void tokenNotFound() throws HandleException {
             final var token = toPbj(KNOWN_TOKEN_WITH_FREEZE);
-            given(tokenStore.getTokenMeta(token)).willReturn(tokenMetaWithFreezeKey());
-            given(accountStore.getAccountById(ACCOUNT_13257))
-                    .willReturn(Account.newBuilder().accountId(ACCOUNT_13257).build());
-            given(tokenStore.get(token)).willReturn(null);
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java:242]: Why were these removed?

Comment (Neeharika-Sompalli) [reply hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java:242]: +1

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAccountWipeHandler.java [hunk 3]
```diff
@@ -229,6 +230,9 @@ private ValidationResult validateSemantics(
         validateTrue(token.wipeKey() != null, ResponseCodeEnum.TOKEN_HAS_NO_WIPE_KEY);
 
         final var accountRel = TokenHandlerHelper.getIfUsable(accountId, tokenId, tokenRelStore);
+        if (token.hasKycKey()) {
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAccountWipeHandler.java:233]: Where is this check in mono-service. I might have missed this 🤔

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java (hunk 2 above)
Comment (natanasow) [reply hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java:242]: The `pureCheck()` method is executed before `tokenStore.getTokenMeta()`, [here](https://github.com/hashgraph/hedera-services/pull/8611/files/5aeeb96b5a4204743e39b3f1e99f335894996d1b#diff-55d2214c26d6ea7346a3bf92e7a3757536c0102dadd73ec12e205e518ed2ae68R65) we can see it. And `pureCheck()` throws `INVALID_TOKEN_ID` exception, [here](https://github.com/hashgraph/hedera-services/pull/8611/files/5aeeb96b5a4204743e39b3f1e99f335894996d1b#diff-55d2214c26d6ea7346a3bf92e7a3757536c0102dadd73ec12e205e518ed2ae68R100). You can track easily who did these changes in develop (adding pure checks). The test started to fail once I updated my branch with develop.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAccountWipeHandler.java (hunk 3 above)
Comment (natanasow) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAccountWipeHandler.java:233]: [This test](https://github.com/hashgraph/hedera-services/pull/8611/files#diff-516656ed4ec511f5fe643accae195e3c0064b054aeb282b129c2b6f545a1facbR215) fails without the check. I accept the tests as a source of truth. I'm not very familiar with the mono repo, but it might be handled [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/models/TokenRelationship.java#L93) because on wipe we're changing balances as well?

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java (hunk 2 above)
Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java:242]: This test is testing that, even though `pureCheck()` passes, we still need to be able to get the token from state (i.e. `tokenStore.get(token)`. Removing these lines invalidates the test

Comment (natanasow) [reply hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java]: Oh, got it :pray:. Just reverted my changes.

**Comment by natanasow** [comment]
@mhess-swl @Neeharika-Sompalli Do I have something else to fix or the pr is ready for merge?

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java (hunk 2 above)
Comment (mhess-swl) [reply hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenUnfreezeAccountHandlerTest.java]: Thank you!

File: hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenGrantKycToAccountHandlerTest.java [hunk 4]
```diff
@@ -221,16 +234,20 @@ void nullAccountIdThrowsException() {
...
+            given(tokenRelStore.get(notNull(), notNull())).willReturn(null);
+            given(expiryValidator.expirationStatus(EntityType.ACCOUNT, false, 0))
+                    .willReturn(OK);
 
             final var txnBody = newTxnBody(true, true);
             given(handleContext.body()).willReturn(txnBody);
             assertThatThrownBy(() -> subject.handle(handleContext))
                     .isInstanceOf(HandleException.class)
-                    .has(responseCode(INVALID_TOKEN_ID));
+                    .has(responseCode(TOKEN_NOT_ASSOCIATED_TO_ACCOUNT));
```
Comment (mhess-swl) [inline hedera-node/hedera-token-service-impl/src/test/java/com/hedera/node/app/service/token/impl/test/handlers/TokenGrantKycToAccountHandlerTest.java:250]: Good catch!

**Review by mhess-swl — DISMISSED** [review]
LGTM. Thanks for the work on this!

**Comment by mhess-swl** [comment]
> @mhess-swl @Neeharika-Sompalli Do I have something else to fix or the pr is ready for merge?

Just approved :)

**Review by Neeharika-Sompalli — DISMISSED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]
